### ISSUE TITLE
cloudwatch_logs output plugin: Could not find sequence token in response: response body is empty

### ISSUE THREAD

**Issue by wtchangdm** [fluent/fluent-bit#3477]
## Bug Report

**Describe the bug**

We are using fluent-bit as our Kubernetes cluster log shipper. Recently, we noticed that some logs are missing on Amazon CloudWatch.
After investigating, I am sure that the missing logs are actually presented on worker node's `/var/log/containers` files.

I then compared the time of logs of missing logs and fluent-bit's logs, there were some errors indicating `Could not find sequence token in response: response body is empty`. One thing that worth mentioning is, it works as expected after fresh start for a while (maybe one day or two).

There was an issue that has the same error reported at https://github.com/aws/aws-for-fluent-bit/issues/161. However, I didn't encounter any crash issue though.

The following logs are fetched after enabling the debug logs:
```
[2021/05/13 03:11:08] [debug] [upstream] KA connection #37 to logs.ap-northeast-1.amazonaws.com:443 has been assigned (recycled)
[2021/05/13 03:11:08] [debug] [http_client] not using http_proxy for header
[2021/05/13 03:11:08] [debug] [aws_credentials] Requesting credentials from the EKS provider..
[2021/05/13 03:11:08] [debug] [http_client] header=POST / HTTP/1.1
Host: logs.ap-northeast-1.amazonaws.com
Content-Length: 4072
User-Agent: aws-fluent-bit-plugin
Content-Type: application/x-amz-json-1.1
X-Amz-Target: Logs_20140328.PutLogEvents
x-amz-date: 20210513T031108Z
x-amz-security-token: <REDACTED>
Authorization: AWS4-HMAC-SHA256 Credential=<REDACTED>, SignedHeaders=content-length;content-type;host;user-agent;x-amz-date;x-amz-security-token;x-amz-target, Signature=<REDACTED>


[2021/05/13 03:11:08] [debug] [output:cloudwatch_logs:cloudwatch_logs.6] PutLogEvents http status=200
[2021/05/13 03:11:08] [debug] [output:cloudwatch_logs:cloudwatch_logs.6] Sent events to <REDACTED_LOG_STREAM>
[2021/05/13 03:11:08] [error] [output:cloudwatch_logs:cloudwatch_logs.6] Could not find sequence token in response: response body is empty
[2021/05/13 03:11:08] [debug] [output:cloudwatch_logs:cloudwatch_logs.6] Sent 2 events to CloudWatch
[2021/05/13 03:11:08] [debug] [socket] could not validate socket status for #37 (don't worry)
```

**To Reproduce**
- Rubular link if applicable: N/A
- Example log message if applicable:
```
{"Timestamp":"2021-05-13T03:24:07.6476101+00:00","Level":"Information","MessageTemplate":"Executing with {Count} rules","RenderedMessage":"Executing with 2 rules","Properties":{"Count":2,"SourceContext":"<REDACTED_STRING>"}}
```
- Steps to reproduce the problem:
  - The setup details are described in the additional context scope below. I can provide the manifests to run  on k8s if needed.

**Expected behavior**

Deliver logs to CloudWatch without intermittent failures.

**Screenshots**

N/A

**Your Environment**

* Version used: 1.7.4
* Configuration:
```
[SERVICE]
    Flush             1
    Log_Level         debug
    Daemon            Off
    Storage.path      /var/log/flb-storage/
    Storage.sync      normal
    Storage.checksum  Off
    Parsers_File      parsers.conf
    HTTP_Server       On
    HTTP_Listen       0.0.0.0
    HTTP_Port         2020

@INCLUDE input-kubernetes.conf
@INCLUDE filter-kubernetes.conf
@INCLUDE output-cloudwatch-logs.conf

[INPUT]
    Name              tail
    Tag_Regex         (?<pod_name>[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*)_(?<namespace_name>[^_]+)_(?<container_name>.+)-(?<docker_id>[a-z0-9]{64})
    Tag               kube__<namespace_name>__<pod_name>__<container_name>__<docker_id>
    Path              /var/log/containers/*.log
    Exclude_Path      /var/log/*fluent-bit*.log,/var/log/*nginx*.log
    Parser            docker
    Storage.type      filesystem
    DB                /var/log/flb_kube.db
    DB.sync           Off
    DB.locking        false
    Buffer_Chunk_Size 10M
    Buffer_Max_Size   15M
    Mem_Buf_Limit     5M
    Skip_Long_Lines   On
    Refresh_Interval  10
    Docker_Mode       On

[FILTER]
    Name                kubernetes
    Match               kube__*
    Kube_Tag_Prefix     ""
    Regex_Parser        kube-parser
    Kube_URL            https://kubernetes.default.svc:443
    Kube_CA_File        /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
    Kube_Token_File     /var/run/secrets/kubernetes.io/serviceaccount/token
    Use_Kubelet         On
    Kubelet_Port        10250
    Merge_Log           On
    Keep_Log            false
    Buffer_Size         1M
    Labels              Off
    Annotations         Off
    tls.verify          Off
    tls.debug           4

[FILTER]
    Name                nest
    Match               kube__*
    Operation           lift
    Nested_under        kubernetes
    Add_prefix          kube_

[FILTER]
    Name                modify
    Match               kube__*
    Rename              kube_pod_name source
    Remove              kube_namespace_name
    Remove              kube_pod_id
    Remove              kube_docker_id
    Remove              kube_host
    Remove              kube_container_hash
    Remove              kube_container_image

[OUTPUT]
    Name                      cloudwatch_logs
    Match                     kube__<REDACTED_NAMESPACE>__*
    region                    ap-northeast-1
    Auto_create_group         On
    Log_group_name            <REDACTED_NAMESPACE>
    Log_stream_name           <REDACTED_NAMESPACE>-via-${HOSTNAME}
```
* Environment name and version (e.g. Kubernetes? What version?): EKS 1.16
* Server type and version: Worker Node AMI:  amazon-eks-node-1.16-v20201211
* Operating System and version: Amazon Linux 2 (`Linux ip-<REDACTED_IP>.ap-northeast-1.compute.internal 4.14.209-160.335.amzn2.x86_64 #1 SMP Wed Dec 2 23:31:46 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`)
* Filters and plugins:
  - kubernetes filter
  - nest filter
  - modify filter
  - cloudwatch_logs output

**Additional context**
I am using [IRSA][1] to give fluent-bit pod permission to send to Amazon CloudWatch. I was suspecting that IRSA might be the reason but since it can still send most of the logs, it doesn't look like an IAM permission issue.

I also run fluent-bit as daemonset with `hostNetowrk` to `true` and `dnsPolicy` as `ClusterFirstWithHostNet ` in order to use kubelet as kubernetes filter's metadata source.

Please let me know if further information is needed. Thanks.

[1]: https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html

**Comment by wtchangdm** [comment]
comment for removing the stale label

**Comment by PettitWesley** [comment]
My fix for this has been released in Fluent Bit 1.7.9: https://github.com/aws/aws-for-fluent-bit/releases/tag/v2.16.0

**Comment by wtchangdm** [comment]
Thank you so much, @PettitWesley! 

Although I am not entirely sure this issue is related to the log loss situation we are facing, I will be upgrading to 1.7.9 and see how it goes.
